### PR TITLE
[DUOS-1826][risk=no] dar status shows open when all elections are closed

### PR DESCRIPTION
**Pull request by shaemarks**
[Link to Jira ticket](https://broadworkbench.atlassian.net/browse/DUOS-1826)
Summary of changes:
- fix bug where incorrect statuses were shown by filtering out RP elections in `determineCollectionStatus`
- Add tests for `determineCollectionStatus` and `processElectionStatus`
- Update string equivalence conditionals to use lowercase version of string 

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: cypress/component/utils/utils.spec.js [hunk 1]
```diff
@@ -17,6 +18,53 @@ const collectionsSkeleton = [
   }
 ];
 
+const collectionWithMixedElectionStatuses = {
+  dars: {
+    0: {
```
Comment (lu-c) [inline cypress/component/utils/utils.spec.js:23]: Why key this by index? Would an array be a better object for `dars`?

File: src/libs/utils.js [hunk 2]
```diff
@@ -66,9 +66,10 @@ export const darCollectionUtils = {
             }
             return [];
           } else {
-            //if elections exist, filer out elections based on relevant ids
+            //if elections exist, filter out elections based on relevant ids
+            //only Data Access elections impact the status of the collection
             //NOTE: Admin does not have relevantIds, DAC roles do
-            const electionArr = Object.values(elections);
+            const electionArr = filter(election => toLower(election.electionType) === 'dataaccess')(Object.values(elections));
```
Comment (lu-c) [inline src/libs/utils.js:72]: LGTM, like the change for lower 👍

**Review by lu-c — APPROVED** [review]
LGTM

File: cypress/component/utils/utils.spec.js (hunk 1 above)
Comment (rushtong) [reply cypress/component/utils/utils.spec.js:23]: Real dar collection data is a map of dar reference id to dar. Similarly, elections are a map of election id to election, etc.

**Review by rushtong — APPROVED** [review]
LGTM 👍🏽

**Pull request merged**